### ISSUE TITLE
Grafana 8: Graphs using MySQL data source don't align to the specified interval in query (like full hour)

### ISSUE THREAD

**Issue by ManuelBauer** [grafana/grafana#36878]
**What happened**:
Since the update to Grafana 8 graphs using the MySQL data source don't align correctly to the specified interval (like full hour) used in the query. The minute and second shown at each data point matches the current minute and second.

As seen in the screenshot, I've loaded the page at 13:44:59. All data points show these as minutes and seconds.
<img width="501" alt="grafana8" src="https://user-images.githubusercontent.com/1709215/126035832-5ad90b44-3fc2-4380-9396-5b41ecdd73ef.png">

Query used:
```
SELECT
$__timeGroup(`timestamp`, '1h', 0) as time,
  count(*) as value,
  uri as metric
FROM pageview
WHERE siteId = 5
group by time, uri
ORDER BY timestamp ASC
```

Generated query by Grafana:
```
SELECT
UNIX_TIMESTAMP(`timestamp`) DIV 3600 * 3600 as time,
  count(*) as value,
  uri as metric
FROM pageview
WHERE siteId = 5 and `timestamp` BETWEEN FROM_UNIXTIME(1626500307) AND FROM_UNIXTIME(1626521907)
group by time, uri
ORDER BY timestamp ASC
```

Example query result:
<img width="470" alt="sql-result" src="https://user-images.githubusercontent.com/1709215/126035870-46836f9b-b666-48cf-a46c-37579fb8e2bb.png">

**What you expected to happen**:
That the timeseries is align to the specified interval. In my case would this be the full hour. In Grafana 7 everything worked fine:
<img width="506" alt="grafana7" src="https://user-images.githubusercontent.com/1709215/126035703-04b53df2-d000-4979-a53a-9206f058f46c.png">

I think the returned time from the query result is not used completely an is replaced partly by the current time.

**How to reproduce it (as minimally and precisely as possible)**:
- Create a demo table like this:
```
CREATE TABLE `pageview` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `siteId` int(11) NOT NULL,
  `timestamp` timestamp NOT NULL DEFAULT '0000-00-00 00:00:00',
  `uri` text NOT NULL,
  PRIMARY KEY (`id`),
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4;
```

- Insert some demo data:
```
INSERT INTO `pageview` (`siteId`, `timestamp`, `uri`)
VALUES
	(5, '2021-07-17 14:19:12', '/page1'),
	(5, '2021-07-17 14:18:59', '/page1'),
	(5, '2021-07-17 14:12:41', '/page1'),
	(5, '2021-07-17 13:02:02', '/page1'),
	(5, '2021-07-17 13:01:06', '/page1'),
	(5, '2021-07-17 12:45:00', '/page1'),
	(5, '2021-07-17 12:44:08', '/page1'),
	(5, '2021-07-17 12:35:16', '/page1'),
	(5, '2021-07-17 12:23:18', '/page1'),
	(5, '2021-07-17 11:13:36', '/page1'),
	(5, '2021-07-17 10:58:14', '/page1'),
	(5, '2021-07-17 10:56:11', '/page1'),
	(5, '2021-07-17 10:55:25', '/page1'),
	(5, '2021-07-17 10:20:54', '/page2'),
	(5, '2021-07-17 10:10:52', '/page3'),
	(5, '2021-07-17 09:44:01', '/page3'),
	(5, '2021-07-17 09:25:44', '/page2'),
	(5, '2021-07-17 09:10:20', '/page1');
```

- Creating a MySQL data source
- Creating a graph using the query above

**Anything else we need to know?**:
Timezone of Server (Grafana / MariaDB) and Browser is Europe/Berlin.

**Environment**:
- Grafana version: 8.0.6
- Data source type & version: MySQL (included version with 8.0.6)
- OS Grafana is installed on: Debian 10.0 (via Docker)
- User OS & Browser: macOS 11.4 / Google Chrome 91
- Database: MariaDB 10.6
- Grafana plugins: -
- Others: -

**Comment by ying-jeanne** [comment]
i am able to reproduce this behavior, it is happening when we put the filled value into timeGroup only, a fix is needed.

**Comment by ying-jeanne** [comment]
the fix of this issue is at https://github.com/grafana/grafana/pull/36937

**Comment by ManuelBauer** [comment]
Thanks for the fix :) I can confirm that with version 8.1.0 the issue no longer exists 👍
